haystack/document_stores/__init__.py

```diff
@@ -7,7 +10,22 @@
 from haystack.document_stores.es_converter import elasticsearch_index_to_document_store
 from haystack.document_stores.es_converter import open_search_index_to_document_store
 
-from haystack.document_stores.elasticsearch import ElasticsearchDocumentStore
+try:
+    # Use appropriate ElasticsearchDocumentStore depending on ES client version
+    from elasticsearch import __version__ as ES_VERSION
+
+    if ES_VERSION[0] == 7:
+        from haystack.document_stores.elasticsearch7 import ElasticsearchDocumentStore  # type: ignore  # pylint: disable=reimported,ungrouped-imports
+    elif ES_VERSION[0] == 8:
+        if not typing.TYPE_CHECKING:
+            from haystack.document_stores.elasticsearch8 import ElasticsearchDocumentStore  # type: ignore  # pylint: disable=reimported,ungrouped-imports
+    else:
+        # Use Elasticsearch 7 as default
+        from haystack.document_stores.elasticsearch7 import ElasticsearchDocumentStore
```

Comment:
This can just be 
```
if ES_VERSION[0] == 8:
    from haystack.document_stores.elasticsearch8 import ElasticsearchDocumentStore # type: ignore # pylint:
else:
    from haystack.document_stores.elasticsearch7 import ElasticsearchDocumentStore # type: ignore # pylint:
```